rclcpp/src/rclcpp/duration.cpp

```diff
@@ -238,11 +246,19 @@ Duration::to_rmw_time() const
     throw std::runtime_error("rmw_time_t cannot be negative");
   }
 
-  // reuse conversion logic from msg creation
-  builtin_interfaces::msg::Duration msg = *this;
+  // Purposefully avoid using the same message creation logic to avoid possible overflow
+  // converting from uint64_t to int32_t, then back to uint64_t
```

Comment:
rcutils_duration_value_t is `int64_t`, im not mistaken?

```suggestion
  // converting from int64_t to int32_t, then back to uint64_t
```

Reply:
we could also write a template for a common implementation, but it doesn't seem to be worth it

Reply:
Updated the comment 05c0169

I realized the logic here could be simplified further since we guarantee that there are no negative values (throwing exception if there are), so I didn't write a template.